feat(compiler-cli): Adapt the compiler to use standalone by default.

JeanMeche opened the pull request:
On top of #58160 & #58175

This PR flips the default for standalone from `false` to `true` for components, directives and pipes. 
This PR also includes the necessary changes for the impacted tests as well as other targets that dependens on them (adev, benchmarks etc).

AndrewKushnir commented on packages/core/src/render3/jit/standalone-default-value.ts:
```diff
@@ -0,0 +1,13 @@
+ * A constant defining the default value for the standalone attribute in Directive and Pipes decorators.
+ * Extracted to a separate file to facilitate G3 patches.
+ */
+export const NG_STANDALONE_DEFAULT_VALUE = true;
```
I think it'd be great to land this change as a refactoring first, without changing the default. After that, we can land another PR that would flip the flag. We'll need those 2 steps to reduce risks of landing this change in g3.

JeanMeche replied:
Great suggestion, It'll split the change even more.

JeanMeche replied:
#58175 will be the intermediate PR.

JeanMeche committed:
feat(core): flipping the default value for `standalone` to `true`

With this commit directives, components & pipes are standalone by default.

To be declared in an `NgModule`, those require now `standalone: false`.

JeanMeche committed:
refactor(compiler): dynamic default for the partial compiler.

Use `semver` in the partial compiler to decide on a default value

Co-authored-by: Alex Rickabaugh <alxhub@users.noreply.github.com>

JeanMeche committed:
refactor(upgrade): update API golden.

`UpgradeComponent` is meant to be used as a subclass, it doesn't need to disable `standalone`.

JeanMeche committed:
refactor(compiler-cli): Update Golden partials

After flipping the standalone default value, this is reflected in the partial output result.

JeanMeche committed:
refactor(compiler-cli): Migrate manually ngtsc tests to standalone by default

This commit is part of the migration to standalone by default.

JeanMeche committed:
refactor(migrations): Port standalone migration to standalone by default

With this commit, the standalone migration will only migrate code with `standalone: false` to standalone by default (without the standalone attribute)

alxhub commented on packages/compiler-cli/linker/src/file_linker/partial_linkers/util.ts:
```diff
@@ -113,3 +114,11 @@ export function extractForwardRef<TExpression>(
   );
 }
+
+const STANDALONE_IS_DEFAULT_RANGE = new semver.Range('>= 19.0.0', {
```
We should make sure that `0.0.0-PLACEHOLDER` is also considered to pass.

JeanMeche committed:
fixup! refactor(compiler): dynamic default for the partial compiler.

JeanMeche commented:
This change has currently an issue in our compliance tests.
Tests that rely on explicit `standalone: false` aren't compiled correctly by the linked target.

JeanMeche committed:
fixup! refactor(compiler): dynamic default for the partial compiler.

JeanMeche committed:
fixup! refactor(compiler-cli): Update Golden partials

jelbourn reviewed (approved):
LGTM

Reviewed-for: global-approvers

jelbourn commented:
Caretaker note: presubmit failure looks like a flake to me; I ran a TGP for these changes yesterday and it was green

devversion commented:
This PR was merged into the repository by commit 9ab663e7a29da63df03f453678ae9ed441c67a5d.

The changes were merged into the following branches: main
